Explorer/Assets/DCL/EmojiPanel/EmojiPanelController.cs

```diff
@@ -0,0 +1,150 @@
+using Cysharp.Threading.Tasks;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using UnityEngine;
+using UnityEngine.UI;
+using Utility;
+using Object = UnityEngine.Object;
+
+namespace DCL.Emoji
+{
+    public class EmojiPanelController
+    {
+        public event Action<string> OnEmojiSelected;
+        private readonly EmojiPanelView view;
+        private readonly EmojiPanelConfigurationSO emojiPanelConfiguration;
+        private readonly EmojiButton emojiButtonPrefab;
+        private readonly EmojiSectionView emojiSectionPrefab;
+        private readonly EmojiSearchController emojiSearchController;
+
+        private readonly List<EmojiSectionView> emojiSectionViews = new ();
+        public readonly Dictionary<string, EmojiData> EmojiNameMapping = new ();
+        private readonly Dictionary<string, string> emojiValueMapping = new ();
+        private readonly Dictionary<EmojiSectionName, RectTransform> sectionTransforms = new ();
+
+        private CancellationTokenSource cts = new CancellationTokenSource();
+
+        private int startDec;
+        private int endDec;
+        private int emojiCode;
+        private string emojiChar;
+
+        public EmojiPanelController(
+            EmojiPanelView view,
+            EmojiPanelConfigurationSO emojiPanelConfiguration,
+            TextAsset emojiMappingJson,
+            EmojiSectionView emojiSectionPrefab,
+            EmojiButton emojiButtonPrefab)
+        {
+            this.view = view;
+            this.emojiPanelConfiguration = emojiPanelConfiguration;
+            this.emojiSectionPrefab = emojiSectionPrefab;
+            this.emojiButtonPrefab = emojiButtonPrefab;
+            emojiSearchController = new EmojiSearchController(view.searchPanelView, view.emojiSearchedContent, emojiButtonPrefab);
+            emojiSearchController.OnSearchTextChanged += OnSearchTextChanged;
+            emojiSearchController.OnEmojiSelected += emoji => OnEmojiSelected?.Invoke(emoji);
+            foreach (var emojiData in JsonConvert.DeserializeObject<Dictionary<string, string>>(emojiMappingJson.text))
+            {
+                EmojiNameMapping.Add(emojiData.Key, new EmojiData($"\\U000{emojiData.Value.ToUpper()}", emojiData.Key));
+                emojiValueMapping.Add(emojiData.Value.ToUpper(), emojiData.Key);
```

Comment:
Can we convert the key to `int` because we access this dict by `int` (formatted to `string`) to avoid intermediate allocations?